src/test/test-time-util.c

```diff
@@ -701,6 +701,17 @@ static void test_parse_timestamp_impl(const char *tz) {
         test_parse_timestamp_one("70-01-01 00:00:01.001 UTC", 0, USEC_PER_SEC + 1000);
         test_parse_timestamp_one("70-01-01 00:00:01.0010 UTC", 0, USEC_PER_SEC + 1000);
 
+        /* The same, but with a T like ISO 8601 */
+        test_parse_timestamp_one("Thu 1970-01-01T00:01 UTC", 0, USEC_PER_MINUTE);
```

Comment:
Also, if we're going to do ISO 8601, then it should also support `20230904T131042Z` and maybe also `2023‐09‐04T20:10:42+07:00`.

Reply:
And allows the timestamp to stop at any field, yes. You clearly don't care about parsing ISO 8601 stamps and I don't really care either, but if you *are* gonna have two formats that's are one byte off of being compatible with ISO 8601, then that's annoying.

This is why I said "like" in the comment – it's a "Systemd-Format Canonical Timestamp But With A `T` Where Users Write It (Users Write It There Because It Looks Like An ISO Timestamp They Write Everywhere Else)".
It's out of scope for me right now to invent full ISO 8601 parsing (and I don't care to hear what lennart would have to say about it this time); this is an ergonomics fix, not a conformance one.